Libs/MRML/Logic/vtkMRMLColorLogic.cxx

```diff
@@ -1076,3 +1094,699 @@ vtkMRMLProceduralColorNode* vtkMRMLColorLogic::CopyProceduralNode(vtkMRMLColorNo
+  std::transform(inputTypeName.begin(), inputTypeName.end(), inputTypeName.begin(), ::tolower);
+  std::transform(inputModifierName.begin(), inputModifierName.end(), inputModifierName.begin(), ::tolower);
+
+  ColorCategorizationMapType::const_iterator iter = this->colorCategorizationMaps[lutName].begin();
+  ColorCategorizationMapType::const_iterator iterEnd = this->colorCategorizationMaps[lutName].end();
+  for (;iter!=iterEnd;iter++)
+    {
+    ColorLabelCategorization mapCat = iter->second;
+
+    // fuzzy comparison rules:
+    //  -- property category is ignored
+    //  -- property type must be found in the mapping
+    //  -- if modifier is non-empty, must match as well
+    //  -- capitalization is ignored
+    //  -- look at the meaning, ignore codes and designators
+    std::string mapTypeName, mapModifierName;
+
+    mapTypeName = mapCat.SegmentedPropertyType.CodeMeaning;
+    mapModifierName = mapCat.SegmentedPropertyTypeModifier.CodeMeaning;
+    std::transform(mapTypeName.begin(), mapTypeName.end(), mapTypeName.begin(), ::tolower);
+    std::transform(mapModifierName.begin(), mapModifierName.end(), mapModifierName.begin(), ::tolower);
+
+    if (mapTypeName.find(inputTypeName) != std::string::npos)
+      {
+      // found match in category name
+      if (inputModifierName != "")
+        {
+        // modifier is not empty
+        if (mapModifierName.find(inputModifierName) != std::string::npos)
+          {
+          labelFound = iter->first;
+          break;
+          }
+        }
+      else
+        {
+        // modifier is empty, and category matches, assume this is the right
+        // term
+        labelFound = iter->first;
+        break;
+        }
+      }
+  }
+
+  if (labelFound != -1)
+    {
+    label = labelFound;
+    }
+  return (labelFound==-1) ? false: true;
+}
+
+//---------------------------------------------------------------------------
+bool vtkMRMLColorLogic::PrintCategorizationFromLabel(int label, const char *lutName)
+{
+  ColorLabelCategorization labelCat;
+  if (!this->TerminologyExists(lutName))
+    {
+    return false;
+    }
+  if (this->colorCategorizationMaps[lutName].find(label) !=
+    this->colorCategorizationMaps[lutName].end())
+    {
+    labelCat = this->colorCategorizationMaps[lutName][label];
+    labelCat.PrintSelf(std::cout);
+    return true;
+    }
+  return false;
+}
+
+//---------------------------------------------------------------------------
+std::string vtkMRMLColorLogic::RemoveLeadAndTrailSpaces(std::string in)
+{
+  std::string ret = in;
+  ret.erase(ret.begin(), std::find_if(ret.begin(),ret.end(),
+    std::not1(std::ptr_fun<int,int>(std::isspace))));
+  ret.erase(std::find_if(ret.rbegin(),ret.rend(),
+    std::not1(std::ptr_fun<int,int>(std::isspace))).base(), ret.end());
+  return ret;
+}
+
+//---------------------------------------------------------------------------
+bool vtkMRMLColorLogic::ParseTerm(std::string str, StandardTerm& term)
```

Comment:
This function should probably be made more robust to the file content. As is, it may crash if the content of the file is not what is expected.